test/integration/builder/additional.js

```diff
@@ -585,6 +585,80 @@ module.exports = function(knex) {
         });
     });
 
+    it('.timeout(ms, {cancel: true}) should throw TimeoutError and cancel slow query in transaction', function() {
+      var dialect = knex.client.dialect;
+      if(dialect === 'sqlite3') { return; } //TODO -- No built-in support for sleeps
+      if (/redshift/.test(dialect)) { return; }
+
+      // There's unexpected behavior caused by knex releasing a connection back
+      // to the pool because of a timeout when a long query is still running.
+      // A subsequent query will acquire the connection (still in-use) and hang
+      // until the first query finishes. Setting a sleep time longer than the
+      // mocha timeout exposes this behavior.
```

Comment:
Did I understood that this test relays on moche timing out test execution? If that was really the it is definitely wrong way to go.

Reply:
I just copied the non-transactional test case and added a transaction wrapper around the execution at the bottom.

This comment is from the original test (added in ea5cbec).

It looks like the behavior was fixed recently in 8771bd4.

If you're interested in moving forward with this PR, I can try sprucing up these tests.